File: java/src/main/java/com/test/java/animals/DogIF.java
package com.test.java.animals;

import com.hubspot.immutables.style.HubSpotStyle;
import org.immutables.value.Value;

import java.util.List;

@Value.Immutable
@HubSpotStyle
public interface DogIF extends Animal {
    @Value.Default
    default String getNoise() {
        return "Woof";
    }

    List<Ball> getBalls();
}

File: kotlin/src/main/kotlin/com/test/kotlin/animals/DogIF.kt
package com.test.kotlin.animals

import com.hubspot.immutables.style.HubSpotStyle
import org.immutables.value.Value

@Value.Immutable
@HubSpotStyle
interface DogIF : Animal {
    @JvmDefault
    override val noise: String
        //@Value.Default
        get() = "Woof"

    val balls: List<Ball>
}
